Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,426 @@
+                }
+            }
+
+            SMcomponent.FinalCountdown = true;
+
+            SMcomponent.DelamTimerAccumulator += frameTime;
+            SMcomponent.SpeakAccumulator += frameTime;
+            int RoundSeconds = SupermatterComponent.DelamTimerTimer - (int)Math.Floor(SMcomponent.DelamTimerAccumulator);
+
+            //we healed out of delam, return
+            if (SMcomponent.DamageArchived < SupermatterComponent.ExplosionPoint)
+            {
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-safe-allert"), "Supermatter", false);
+                SMcomponent.FinalCountdown = false;
+                return;
+            }
+            //we're more than 5 seconds from delam, only yell every 5 seconds.
+            else if (RoundSeconds >= 5 && SMcomponent.SpeakAccumulator >= 5)
+            {
+                SMcomponent.SpeakAccumulator -= 5;
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-seconds-before-delam", ("Seconds", RoundSeconds)), "Supermatter", false);
+            }
+            //less than 5 seconds to delam, count every second.
+            else if (RoundSeconds <  5 && SMcomponent.SpeakAccumulator >= 1)
+            {
+                SMcomponent.SpeakAccumulator -= 1;
+                _chatManager.DispatchStationAnnouncement(Loc.GetString("supermatter-seconds-before-delam", ("Seconds", RoundSeconds)), "Supermatter", false);
+            }
+
+            //play an alarm as long as you're delaming
+            if (SMcomponent.FinalCountdown)
+            {
+                SoundSystem.Play(Filter.Pvs(Uid, 5), SMcomponent.DelamAlarm.GetSound(), Uid, audioParams);
+            }
+
+            //TODO: make tesla(?) spawn at SupermatterComponent.PowerPenaltyThreshold and think up other delam types
+            //times up, explode or make a singulo
+            if (SMcomponent.DelamTimerAccumulator >= SupermatterComponent.DelamTimerTimer)
+            {
+                if (_delamType.Equals(SupermatterComponent.DelamType.Singulo))
+                {
+                    //spawn a singulo :)
+                    EntityManager.SpawnEntity("Singularity", Xform.Coordinates);
+                }
+                else if (_delamType.Equals(SupermatterComponent.DelamType.Explosion))
+                {
+                    //esplosion!!!!!
+                    _explosion.TriggerExplosive(Uid, explosive: Xplode, totalIntensity: 500000, radius: 500, user: Uid);
+                }
+
+                SMcomponent.FinalCountdown = false;
+                return;
+            }
+        }
+
+        /// </summary>
+        /// Determines if an entity can be dusted
+        /// </summary>
+        public bool CannotDestroy(EntityUid Uid)
+        {
+            bool Tag = false;
+            bool Static = false;
+
+            if (EntityManager.HasComponent<TagComponent>(Uid))
+            {
+                Tag = _tag.HasTag(Uid, "SMimmune");
+            }
```

Comment:
That check is unnecessary. `HasTag` uses resolve and returns false if an entity doesn't have the tag component.